lib/grandpa/grandpa.go

```diff
@@ -199,16 +199,18 @@ func (s *Service) Stop() error {
 }
 
 // authorities returns the current grandpa authorities
-func (s *Service) authorities() []*types.Authority {
-	ad := make([]*types.Authority, len(s.state.voters))
-	for i := 0; i < len(s.state.voters); i++ {
-		ad[i] = &types.Authority{
-			Key:    &s.state.voters[i].Key,
-			Weight: s.state.voters[i].ID,
+func (s *Service) authorityKeySet() (authorityKeys map[string]*types.Authority) {
```

Comment:
As mentioned in https://github.com/ChainSafe/gossamer/issues/2960 we want a **set** so no value in the map. Change `map[string]*types.Authority` to `map[string]struct{}`. This is because we never use the map values, we only care about the key being present or not. Also `struct{}{}` uses 0 byte unlike any other type.

FYI to set a key in such map 'set', use `authorityKeys[string(encodedKey)] = struct{}{}`